pkg/test/istio_io/yaml_step.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio_io
+
+import (
+	"path"
+
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+)
+
+var _ step = &yamlStep{}
+
+type yamlStep struct {
+	f         FileSelector
+	namespace string
+	delete    bool
+}
+
+func newYamlStep(namespace string, f FileSelector, delete bool) step {
+	return &yamlStep{
+		f:         f,
+		namespace: namespace,
+		delete:    delete,
+	}
+}
+
+func (s *yamlStep) Run(ctx Context) {
+	yamlFile := s.f.SelectFile(ctx)
```

Comment:
Can we use better variable names than s and f? 

s.f.SelectFile increases cognitive load as the reader has to figure out what both s and f mean. Can we say "step.file"?